Orange/widgets/visualize/owscatterplot.py

```diff
@@ -428,25 +428,35 @@ def selection_changed(self):
         self.commit()
 
     def send_data(self):
-        selected = None
-        selection = None
-        # TODO: Implement selection for sql data
+        def _get_selected():
+            if not len(selection):
+                return None
+            groups = graph.selection[selection, np.newaxis]
+            group_var = DiscreteVariable(
+                name=get_next_name(domain, "Group"),
+                values=["G{}".format(i + 1) for i in range(np.max(groups))])
+            sel_domain = Domain(
+                domain.attributes, domain.class_vars,
+                domain.metas + (group_var, ))
+            metas = np.hstack((self.data.metas[selection], groups - 1))
+            return Table(
+                sel_domain, data.X[selection], data.Y[selection], metas)
```

Comment:
Wondering, could you somehow reuse/extend `create_annotated_table()` for this?

Reply:
I started by filtering the annotated data, but it was messy. Now I've changed the `create_groups_table` and it's way simpler. Thanks for the suggestion.